diff --git a/node_modules/esbuild/lib/main.js b/node_modules/esbuild/lib/main.js
index 49721d1..22fdbe2 100644
--- a/node_modules/esbuild/lib/main.js
+++ b/node_modules/esbuild/lib/main.js
@@ -1845,7 +1845,13 @@ by esbuild to install the correct binary executable for your current platform.`)
 }
 
 // lib/npm/node.ts
-var child_process = require("child_process");
+// BUN NOTE: polyfill child_process with bun-utilities
+var child_process;
+if (process.versions.bun) {
+  child_process = require('bun-utilities');
+} else {
+  child_process = require("child_process");
+}
 var crypto = require("crypto");
 var path2 = require("path");
 var fs2 = require("fs");
@@ -2038,11 +2044,24 @@ var ensureServiceIsRunning = () => {
   if (longLivedService)
     return longLivedService;
   let [command, args] = esbuildCommandAndArgs();
-  let child = child_process.spawn(command, args.concat(`--service=${"0.14.48"}`, "--ping"), {
-    windowsHide: true,
-    stdio: ["pipe", "pipe", "inherit"],
-    cwd: defaultWD
-  });
+  // BUN NOTE: Use bun-utilities compatible API
+  let child
+  if (process.versions.bun) {
+    child = child_process.spawn(command, args.concat(`--service=${"0.14.48"}`, "--ping"), {
+      // windowsHide: true,
+      // stdio: ["pipe", "pipe", "inherit"],
+      stdin: "pipe",
+      stdout: "pipe",
+      stderr: "inherit",
+      cwd: defaultWD
+    });
+  } else {
+    child = child_process.spawn(command, args.concat(`--service=${"0.14.48"}`, "--ping"), {
+      windowsHide: true,
+      stdio: ["pipe", "pipe", "inherit"],
+      cwd: defaultWD
+    });
+  }
   let { readFromStdout, afterClose, service } = createChannel({
     writeToStdin(bytes) {
       child.stdin.write(bytes, (err) => {
@@ -2152,12 +2171,23 @@ var runServiceSync = (callback) => {
     esbuild: node_exports
   });
   callback(service);
-  let stdout = child_process.execFileSync(command, args.concat(`--service=${"0.14.48"}`), {
-    cwd: defaultWD,
-    windowsHide: true,
-    input: stdin,
-    maxBuffer: +process.env.ESBUILD_MAX_BUFFER || 16 * 1024 * 1024
-  });
+  // BUN NOTE: Use bun-utilities compatible API
+  let stdout;
+  if (process.versions.bun) {
+    stdout = child_process.exec(command, args.concat(`--service=${"0.14.48"}`), {
+      cwd: defaultWD,
+      // windowsHide: true,
+      // input: stdin,
+      // maxBuffer: +process.env.ESBUILD_MAX_BUFFER || 16 * 1024 * 1024
+    });
+  } else {
+    stdout = child_process.exec(command, args.concat(`--service=${"0.14.48"}`), {
+      cwd: defaultWD,
+      windowsHide: true,
+      input: stdin,
+      maxBuffer: +process.env.ESBUILD_MAX_BUFFER || 16 * 1024 * 1024
+    });
+  }
   readFromStdout(stdout);
   afterClose(null);
 };
